src/components/TodoInput/TodoInput.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect, useRef } from 'react';
+
+import { useTodos } from '../../hooks/useTodos';
+import { USER_ID, addTodo } from '../../api/todos';
+
+const TodoInput: React.FC = () => {
+  const [isAllCompleted, setIsAllCompleted] = React.useState(false);
+
+  const {
+    todos,
+    setTodos,
+    setError,
+    isLoading,
+    setIsLoading,
+    setTempTodo,
+    handleError,
+    isAllDeleted,
+  } = useTodos();
+
+  const [newTodoTitle, setNewTodoTitle] = React.useState('');
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isLoading, isAllDeleted]);
+
+  const hangleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTodoTitle(event.target.value);
+    setError('');
+  };
+
+  const handleAddTodo = () => {
+    if (!newTodoTitle.trim()) {
+      handleError('Title should not be empty');
+
+      return;
+    }
+
+    setTempTodo({
+      id: 0,
+      userId: USER_ID,
+      title: newTodoTitle.trim(),
+      completed: false,
+    });
+
+    const newTodo = {
+      userId: USER_ID,
+      title: newTodoTitle.trim(),
+      completed: false,
+    };
+
+    setIsLoading(true);
+
+    addTodo(newTodo)
+      .then(data => {
+        setTodos(prevTodos => [...prevTodos, data]);
+        setNewTodoTitle('');
+      })
+      .catch(() => {
+        handleError('Unable to add a todo');
+      })
+      .finally(() => {
+        setTempTodo(null);
+        setIsLoading(false);
+      });
+  };
+
+  const handleEnterEvent = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter') {
+      event.preventDefault();
+      handleAddTodo();
+    }
+  };
+
+  const handleToggleAll = () => {
+    setTodos(todos.map(todo => ({ ...todo, completed: !isAllCompleted })));
+    setIsAllCompleted(!isAllCompleted);
```

Comment:
When you need to change state depends on the previous one, use functional `setState`

Reply:
> When you need to change state depends on the previous one, use functional `setState`

Not fixed
Also, looks like this state is redundant, as this value should be dynamic